src/services/classes/Contract.js

```diff
@@ -35,6 +35,19 @@ class Contract extends BcThing {
           if (!deployedCode) throw new Error(`Missing deployed code for contract: ${this.address}`)
           let info = await this.parser.getContractInfo(deployedCode, contract)
           let { interfaces, methods } = info
+
+          if (!interfaces.length) { // if no interfaces... double check
+            const proxyCheckResult = await this.parser.getContractInfoIfProxy(this.address)
+            // if proxy detected, the implementation contract interfaces are used
+            if (proxyCheckResult && proxyCheckResult.interfaces.length) {
+              interfaces = proxyCheckResult.interfaces
+            }
+
+            if (proxyCheckResult && proxyCheckResult.methods.length) {
+              methods = proxyCheckResult.methods
+            }
+          };
```

Comment:
Following the review on https://github.com/rsksmart/rsk-contract-parser/pull/29

```suggestion
          if (!interfaces.includes(ERC1967)) {
            this.setData({ proxyContractInfo: await this.parser. getEIP1967Info(this.address) })
          };
```

Reply:
Ah don't forget to add the interfaces :) 

```js
if (interfaces.length) this.setData({ contractInterfaces: interfaces })
```

Reply:
This should go inside `if (interfaces.length)`